### DISCUSSION TITLE
Cannot get multimap.conf to work as override

### DISCUSSION THREAD

**Discussion by robertklep** [docker-mailserver/docker-mailserver#3613] [Q&A] [answered]
I'd like to add a multimap configuration, specifically to create a recipient blacklist symbol.

I'm using this setup:
```
# config/rspamd/override.d/multimap.conf
LOCAL_BL_RCPT {
  type = "rcpt";
  map = "$LOCAL_CONFDIR/override.d/local_bl_rcpt.map.inc";
  description = "Local rcpt blacklist";
  score = 3;
}

# config/rspamd/override.d/local_bl_rcpt.map.inc
user@example.com
```

I've also explicitly enabled the `multimap` module in `custom-commands.conf`, just in case.

But this doesn't seem to work. I'm expecting the symbol `LOCAL_BL_RCPT` to show up in the list of symbols in the web UI, but it doesn't appear. I can find the files inside the container in `/tmp/docker-mailserver/rspamd/override.d`.

**Comment by OE2WNL** [comment]
are you sure that the variable $LOCAL_CONFDIR is defined in this context?

I'm doing the same as you but I have defined the paths according to the other config files like:

```shell
WHITELIST_IP {
    type = "ip";
    map = "/tmp/docker-mailserver/rspamd/whitelist_ip.map";
    description = "Local ip whitelist";
    action = "accept";
}
```

**Comment by robertklep** [comment]
@OE2WNL good point, I also figured I might have to use the full path but no luck 🙁

Just for my understanding: do I have to do anything special after adding these files and restarting the container?

**Comment by OE2WNL** [comment, marked as answer]
in my dms config folder at host I have the folders
.../config/rspamd/override.d/
with the file multimap.conf

```shell
WHITELIST_IP {
    type = "ip";
    map = "/tmp/docker-mailserver/rspamd/whitelist_ip.map";
    description = "Local ip whitelist";
    action = "accept";
}

WHITELIST_FROM {
        type = "from";
        map = "/tmp/docker-mailserver/rspamd/whitelist_from.map";
        description = "Local from whitelist";
        action = "accept";
}

BLACKLIST_IP {
        type = "ip";
        map = "/tmp/docker-mailserver/rspamd/blacklist_ip.map";
        description = "Local ip blacklist";
        action = "reject";
}

BLACKLIST_FROM {
        type = "from";
        map = "/tmp/docker-mailserver/rspamd/blacklist_from.map";
        description = "Local from blacklist";
        action = "reject";
}
``` 

And when I restart the docker container (`docker compose down`, `docker compose up -d`) and try a testmail from a server that's IP is in my `whitelist_ip.map` file the info in the rspamd webui points out

```
WHITELIST_IP (0) [a.b.c.d]
``` 
as a result of the whitelisting as expected

**Comment by robertklep** [comment]
Can you also find the symbols in the web UI in the "Symbols" table?

**Comment by OE2WNL** [comment]
yes

![grafik](https://github.com/docker-mailserver/docker-mailserver/assets/27812067/ba3ee5c0-d896-42b3-b77d-7515b13c13cc)

![grafik](https://github.com/docker-mailserver/docker-mailserver/assets/27812067/d7216329-6aaa-4625-898e-2209b365514f)

**Comment by OE2WNL** [comment]
do you see entries in the Configuration Tab of you rspamd web-ui?

![grafik](https://github.com/docker-mailserver/docker-mailserver/assets/27812067/c69045d2-cac0-46ed-82be-d2aa64825bba)

so you know that your config files are part of the "active" config

**Comment by robertklep** [comment]
Okay, I'm not sure exactly what changed, but it's working now! I basically copied your configuration and made the relevant changes for my use case.

Thanks!

**Reply by OE2WNL** [reply]
great to here that my technican aura works long distanct ;-)
